x-pack/plugins/ingest_pipelines/public/application/services/breadcrumbs.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { BASE_PATH } from '../../../common/constants';
+import { ManagementAppMountParams } from '../../../../../../src/plugins/management/public';
+
+type SetBreadcrumbs = ManagementAppMountParams['setBreadcrumbs'];
+
+export class BreadcrumbService {
+  private breadcrumbs: {
+    [key: string]: Array<{
+      text: string;
+      href?: string;
+    }>;
+  } = {
+    home: [],
+  };
+  private setBreadcrumbsHandler?: SetBreadcrumbs;
+
+  public setup(setBreadcrumbsHandler: SetBreadcrumbs): void {
+    const homeBreadcrumbText = i18n.translate('xpack.ingestPipelines.breadcrumb.pipelinesLabel', {
+      defaultMessage: 'Ingest Pipelines',
+    });
+
+    this.setBreadcrumbsHandler = setBreadcrumbsHandler;
+
+    this.breadcrumbs.home = [
```

Comment:
These declarations do not have to happen at setup time, it looks like they are not derived from passed in state so perhaps we can declare them on line 19. Wdyt?

Reply:
Then setup would just be

```
this.setBreadcrumbsHandler = setBreadcrumbsHandler;
```

Reply:
Great catch! Fixed.